onejs/index.ts

```diff
@@ -53,76 +72,73 @@ export function useEventfulState<T, K extends keyof T>(obj: T, propertyName: K,
  * @param dependencies: The dependencies to pass to useEffect. Previous versions
  * of the callback will be cleaned up any dependency changes.
  */
-export function useEvent<TEventName extends string, TCallback>(
-    obj: HasCSharpEventBase<TEventName, TCallback>,
-    eventName: TEventName,
-    callback: TCallback,
-    dependencies: any[] = []
+export function useEvent<
+  T extends {
+    [k in `add_${E}` | `remove_${E}`]: (handler: (...args: any) => any) => void
+  },
+  E extends string
+>(
+  obj: T,
+  eventName: E,
+  callback: InferEventHandler<T[`add_${E}`]> &
+    InferEventHandler<T[`remove_${E}`]>,
+  dependencies: any[] = []
 ) {
-    const remove = obj[`remove_${eventName}`].bind(obj)
-    function removeHandler() {
-        remove(callback)
-    }
+  function removeHandler() {
+    obj[`remove_${eventName}`](callback)
+  }
 
-    useEffect(() => {
-        obj[`add_${eventName}`](callback)
-        onEngineReload(removeHandler)
-        return () => {
-            removeHandler()
-            unregisterOnEngineReload(removeHandler)
-        }
-    }, dependencies)
+  useEffect(() => {
+    obj[`add_${eventName}`](callback)
+    onEngineReload(removeHandler)
+    return () => {
+      removeHandler()
+      unregisterOnEngineReload(removeHandler)
+    }
+  }, dependencies)
 }
 
 /**
- * Describes a C# class or struct that contains a property that is a C# event.
- * For example, given a C# class that declares an event called `OnValueChanged`
- * whose delegates accept a single parameter of type `int`, you can declare the
- * type as follows:
- *
- *   type MyType = HasCsharpEvent<"OnValueChanged", number> & {
- *     // other properties...
- *   }
- * 
- * For event delegates that take more than one value, see related types
- * `HasCsharpEvent2` and `HasCsharpEvent3`.
+ * Similar to useEvent() but accepts a ref that references the C# object instead.
+ * Run a callback when a C# event fires. The callback will be cleaned up on
+ * unmount and when the OneJS engine reloads.
+ * @param ref The ref that contains the C# object.
+ * @param eventName The variable name of the C# event.
+ * @param callback The callback to run when the event fires.
+ * @param dependencies: The dependencies to pass to useEffect. Previous versions
+ * of the callback will be cleaned up any dependency changes.
  */
-export type HasCsharpEvent<EventName extends string, TVal> = HasCSharpEventBase<
```

Comment:
Is there a way we can avoid removing these types? These are exported, and will break our code that uses it. We are using these types to describe C# interop classes that expose events to JS, exactly the way the comment describes.

Reply:
Oh yes sorry. I didn't know they were removed. I added them back in with #[cbbfc0a](https://github.com/DragonGround/ScriptLib/commit/cbbfc0a9dbf16f892d4129ef0bcd28c623444e9c)